### ISSUE TITLE
wrong indentation: SCSS map functions with !default; 

### ISSUE THREAD

**Issue by liubiantao** [prettier/prettier#2799]
**Prettier Version:** 1.6.1 / master

**Code**

```css
$theme-checkbox-colors: (
  default: $theme-color-border,
  checked: $theme-color-checked,
  disabled: $theme-color-disabled,
  disabled-font: $theme-color-font-secondary,
) !default;
```

**Expected behavior:**

```css
$theme-checkbox-colors: (
  default: $theme-color-border,
  checked: $theme-color-checked,
  disabled: $theme-color-disabled,
  disabled-font: $theme-color-font-secondary,
) !default;
```

**Actual behavior:**

```css
$theme-checkbox-colors: (
    default: $theme-color-border,
    checked: $theme-color-checked,
    disabled: $theme-color-disabled,
    disabled-font: $theme-color-font-secondary
  )
  !default;
```

and stylelint will throw this error:
```
 38:3  ✖  Expected single space before "!"   declaration-bang-space-before
 42:5  ✖  Expected indentation of 2 spaces   indentation
```

I also found the usage in bootstrap:

```css
$grid-breakpoints: (
  xs: 0,
  sm: 576px,
  md: 768px,
  lg: 992px,
  xl: 1200px
) !default;
```
> https://github.com/twbs/bootstrap/blob/f7eb3c98b261930270c2f65c0e565c454d6cb647/scss/_variables.scss#L145-L151

**Comment by lydell** [comment]
Given that it formats like the expected case when you remove `!default` (or change to `!important`), I consider this a bug.

**Comment by JasonMSims** [comment]
I've stumbled onto this same error as well.

**Comment by hudochenkov** [comment]
In similar situation valid CSS with `!important` works correctly.

**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDABMAOlLMBGqYA1gOYBOEArlACZLoA2AllHKuQLQWq3MIwAFDAjp8EGCIC2AGnQBiVPjC04AM3QAGAKRzFy1RoBMOvUpXr0AZlMLzh9ABZb+ixoCMmnQEp0AQmYpAAcIchhUWABuHABfEBkQCCCYZmgAZ2RQDkoAdwAFDgQMlFQANwhmWni0NJhkNVRGNLgE-HIiYjgYAGUgolZSZBhyKhaQWggwesbmhNZmsLz20ilUaaaxgCs0gA8AIXaSLu7UKTgAGVY4ddnEqhgg+6Mbsb7yBeRwNIyEoPJWGAAdUqMAAFsgAByaX6UZqA9pBT5-OALUrXBLkOAARyozExS1QKzWSAaGwSzSkzCGIzGaQGjDgAEUqBJriSZmNwvhgbQwcgjAlhqhmCwoKQAMIQKSrT5QaDokBUZoAFSUxVJzRiMSAA)
```sh
--parser css
```

**Input:**
```css
a {
	background: linear-gradient(to bottom, #abcdef 0%, #abcdef 20%, #abcdef 30%, #abcdef 40%, #abcdef 100%) !important;
}
```

**Output:**
```css
a {
  background: linear-gradient(
    to bottom,
    #abcdef 0%,
    #abcdef 20%,
    #abcdef 30%,
    #abcdef 40%,
    #abcdef 100%
  ) !important;
}

```

With anything different from `!important` output changes:

**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDABMAOlLMBGqYA1gOYBOEArlACZLoA2AllHKuQLQWq3MIwAFDAjp8EGCIC2AGnQBiVPjC04AM3QAGAKRzFy1RoBMOvUpXr0AZlMLzh9ABZb+ixoCMmnQEp0AQmYpAAcIchhUKAAVAG4cAF8QGRAIIJhmaABnZFAOSgB3AAUOBCyUVAA3CGZaRLQMmGQ1VEYMuCT8ciJiOBgAZSCiVlJkGHIqNpBaCDBG5tak1lawgs7SKVRZlomAKwyADwAhTpIe3tQpOAAZVjhN+eSqGCDHozuJgfIl5HAMrKSg8isGAAdWqMAAFsgAByaf6UVrAzpBb4AuBLcq3JLkOAARyozGxK1Qaw2SCaWySrSkzBGYwmGSGjDgAEUqBJbmS5hNwvhQbQIcgjElRqhmCwoKQAMIQKTrb5QaCYkBUVqRJSlcmtOJxIA)
```sh
--parser css
```

**Input:**
```css
a {
	background: linear-gradient(to bottom, #abcdef 0%, #abcdef 20%, #abcdef 30%, #abcdef 40%, #abcdef 100%) !importanT;
}
```

**Output:**
```css
a {
  background: linear-gradient(
      to bottom,
      #abcdef 0%,
      #abcdef 20%,
      #abcdef 30%,
      #abcdef 40%,
      #abcdef 100%
    )
    !importanT;
}

```

**Comment by azz** [comment]
Fixed in #3724
